bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableUtils.java

```diff
@@ -96,6 +96,8 @@ public static Long hash(Object obj, Node parent) {
             } else {
                 return (long) obj.hashCode();
             }
+        } else if (obj instanceof Long) {
```

Comment:
I suspect that the same thing can happen for the floats also. Shall we add test cases for all basic data types? internally it seems the same algo is used for java double values to calculate the hashcode